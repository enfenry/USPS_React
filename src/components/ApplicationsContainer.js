"use strict"

import * as applicationsActions from '../actions/applicationsActions';
import ApplicationsRender from './ApplicationsRender';
import PropTypes from 'prop-types';
import React, {useEffect} from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

const ApplicationsContainer = (props) => {

    useEffect(() => {
       const { actions } = props;
       actions.readApplications();
    }, [] );

    console.log("applications:", props);

    if (props.applicationRequestState.applicationsReadPending || props.productRequestState.productsReadPending || props.customerRequestState.customersReadPending || props.addressRequestState.addressesReadPending) {

        return (
            <div className="d-flex justify-content-center">
                <div className="spinner-border" role="status">
                    <span className="sr-only">Loading...</span>
                </div>
            </div>
        );
    }
    if (props.applicationRequestState.applicationsReadFailed || props.productRequestState.productsReadFailed || props.customerRequestState.customersReadFailed || props.addressRequestState.addressesReadFailed) {
        return (
            <div className="alert alert-danger" role="alert">
                Error while loading entities!
            </div>
        );

    }
    if ((props.applicationRequestState.applicationsReadSuccess || props.applicationRequestState.applicationsUpdateSuccess || props.applicationRequestState.applicationsCreateSuccess|| props.applicationRequestState.applicationsToOrderSuccess) && props.productRequestState.productsReadSuccess && props.customerRequestState.customersReadSuccess && props.addressRequestState.addressesReadSuccess) {
        return (
            <div className="reactive-margin">
                <ApplicationsRender
                    applications={props.applications}
                    handleUpdate={(values, application) => {
                        props.actions.updateApplication(values, application.ss_applicationid)
                    }}
                    handleDelete={application => {
                        props.actions.deleteApplication(application.ss_applicationid)
                    }}
                    handleCreate={(values) => {
                        props.actions.createApplication(values)
                    }}
                    handleAppToOrder={(application) => {
                            props.actions.applicationToOrder(application.ss_applicationid)
                    }}
                />
            </div>
        );
    } 
    else {
        return (
            <div className="alert alert-danger" role="alert">
                Invalid state! This message should never appear.
            </div>
        );
    }
}


ApplicationsContainer.propTypes = {
    actions: PropTypes.object
};

function mapStateToProps(state) {
    return {
        applications: state.applicationsReducer.applications,
        products: state.productsReducer.products,
        orders: state.ordersReducer.orders,
        customers: state.customersReducer.customers,
        addresses: state.addressesReducer.addresses,
        
        applicationRequestState: state.applicationsReducer.requestState,
        productRequestState: state.productsReducer.requestState,
        addressRequestState: state.addressesReducer.requestState,
        customerRequestState: state.customersReducer.requestState,
    }
}

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators(applicationsActions, dispatch),
    }
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(ApplicationsContainer);