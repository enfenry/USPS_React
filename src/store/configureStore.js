import { createStore, applyMiddleware, compose } from 'redux';
import reduxImmutableStateInvariant from 'redux-immutable-state-invariant';
import thunk from 'redux-thunk';
import rootReducer from '../reducers';

import { createLogger } from 'redux-logger'
import DevTools from '../containers/DevTools';

function configureStore(initialState) {

  const logger = createLogger()

  const middlewares = [
    reduxImmutableStateInvariant(),
    thunk,
    logger
  ];

  const store = createStore(
    rootReducer,
    initialState,
    compose(
      applyMiddleware(...middlewares),
      DevTools.instrument()
    )
  );

  return store;
}

export default configureStore;
