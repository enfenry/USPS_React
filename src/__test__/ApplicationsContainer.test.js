import React from 'react';
import ApplicationsContainer from '../components/ApplicationsContainer';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import * as basicStateUtils from "./basicState"; 

import configureStore from 'redux-mock-store';
const mockStore = configureStore([]);

test('ApplicationsContainer renders correctly', () => {

  const store = mockStore(basicStateUtils.getBasicState());

  const component = renderer.create(
    <Provider store={store}>
      <ApplicationsContainer />
    </Provider>
  );

  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});