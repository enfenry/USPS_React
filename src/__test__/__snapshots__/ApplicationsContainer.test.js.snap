// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationsContainer renders correctly 1`] = `
<div
  className="m-5"
>
  <div>
    <h1>
      Applications
    </h1>
    <table
      className="table"
    >
      <thead>
        <tr>
          <th
            scope="col"
          >
            Name
          </th>
          <th
            scope="col"
          >
            ID
          </th>
          <th
            scope="col"
          >
            Type
          </th>
          <th
            scope="col"
          >
            Created On
          </th>
          <th
            scope="col"
          />
          <th
            scope="col"
          />
          <th
            scope="col"
          />
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            example name
          </td>
          <td>
            example id
          </td>
          <td>
            example application type
          </td>
          <td>
            example date
          </td>
          <td
            align="right"
          >
            <div>
              <p>
                <button
                  aria-label={null}
                  className="btn btn-secondary btn btn-secondary"
                  onClick={[Function]}
                  type="button"
                >
                  View
                </button>
              </p>
            </div>
          </td>
          <td
            align="right"
          >
            <div>
              <p>
                <button
                  aria-label={null}
                  className="btn btn-success btn btn-secondary"
                  onClick={[Function]}
                  type="button"
                >
                  Update
                </button>
              </p>
            </div>
          </td>
          <td
            align="right"
          >
            <div>
              <p>
                <button
                  aria-label={null}
                  className="btn btn-danger btn btn-secondary"
                  onClick={[Function]}
                  type="button"
                >
                  Delete
                </button>
              </p>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
    <button
      aria-label={null}
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      Add
    </button>
  </div>
</div>
`;
